nixos/modules/services/backup/tarsnap.nix

```diff
@@ -242,9 +254,11 @@ in
 
     systemd.services."tarsnap@" = {
       description = "Tarsnap archive '%i'";
-      requires    = [ "network.target" ];
+      requires    = [ "network-online.target" ];
+      after       = [ "network-online.target" ];
 
-      path = [ pkgs.tarsnap pkgs.coreutils ];
+      path = [ pkgs.tarsnap pkgs.coreutils pkgs.iputils ];
+      preStart = "while ! ping -q -c 1 betatest-server.tarsnap.com &> /dev/null; do true; done";
```

Comment:
`betatest-server.tarsnap.com` is the address the tarsnap client connects to, i.e, it's where the tarsnap server code is running. Why it still carries the name `betatest` after all these years is unknown to me (^_^).

The goal here is to wait until we can connect to the server so that tarsnap doesn't run and just die because it couldn't connect to the server when it could have connected successfully a few seconds later (after the network has finished setting up).